### PR TITLE
[core] Make DatePickers tests independent from other packages

### PR DESCRIPTION
**Pull request by JCQuintas**
Related to https://github.com/mui/mui-x/issues/13882

- Created `test/unit-date-pickers` which holds all config and helpers for date pickers unit testing
- Created `test/shared` for any shared functionality that should happen in most/all tests
- Long term goal is to have no files in the root `test/`, only dirs
- Also will move all `treeview|data-grid|charts` test configs into their own folders/scripts

### CODE REVIEW

**Review by LukasTy — COMMENTED** [review]
Thanks for your effort! 🙏 
Leaving some initial feedback.

File: test/unit-date-pickers/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "@mui-x-internal/unit-date-pickers",
+  "version": "7.11.0",
+  "private": true,
+  "scripts": {
+    "typescript": "tsc -p tsconfig.json",
+    "test:unit": "cross-env NODE_ENV=test TZ=UTC mocha -n expose_gc --config .mocharc.js"
```
[the remark this answers is not in the document]
Comment (JCQuintas) [reply test/unit-date-pickers/package.json:7]: My first approach was to have the config on each package. 😅
It works, but I felt it to be a bit clunky. `x-date-pickers` and `x-date-pickers-pro` use exactly the same config, so they would be duplicated or we would need a place to store them. Should they be in `x-date-pickers`? Felt a bit odd, as there might be necessary types/imports from `-pro` in some cases.

- `test/unit-date-pickers` -> depends on
  - `@mui/x-date-pickers`
  - `@mui/x-date-pickers-pro`

This makes the tests dependant on the packages. Often times you will want to run the tests for them together anyways, so I felt this was fine.

The problem with utils folder is that it is a too generic name, so I would rather we move away from it in this use-case. We(as humans) tend to put anything there, so it acts as a container for everything, which is almost the same as no container at all. In general, folders that have a semantical meaning are easier to navigate, and should be preferred.

It is also important to note these changes need to be done step by step. As we need to keep CI and coverage working as we go 😅 

So in short:

- Either duplicate config, or shared, so moved all config to a single place, which made easier to just have a single config.
- Wanted to have a specific place for the `date-picker` utils, moved them to same package.
- Changes don't affect CI/coverage for now

[the remark this answers is not in the document]
Comment (JCQuintas) [reply test/unit-date-pickers/package.json:7]: > I'm not the biggest fan of this direction as we'd end up with duplicate testing packages for each product line. 🙈

I don't get what you mean with this. Which packages would be we duplicating?

File: babel.config.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ const defaultAlias = {
   ),
   docs: resolveAliasPath('./node_modules/@mui/monorepo/docs'),
   test: resolveAliasPath('./test'),
+  '@unit/date-pickers': resolveAliasPath('./test/unit-date-pickers'),
```
[the remark this answers is not in the document]
Comment (JCQuintas) [reply babel.config.js:28]: I don't mind much about the naming. I just used what felt best. `test` is generic so it didn't give much value, eg: `test/date-pickers` could be confusing if we have `e2e-date-pickers` in the future. `test/pickers` could be confusing if we have a `color-picker` which is a different package. In general (contradictorily), I find the more precise the better 😅 

So I felt `@unit/date-pickers` to be the most concise and gives all the info necessary. We can def change it to `test/unit/date-pickers` or `test/pickers` or anything else if you don't think the considerations above are relevant. This is quite easy to refactor, we could change at any time if we decide to go with `test/pickers` and later we have something else there

File: test/unit-date-pickers/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexfauquette) [reply test/unit-date-pickers/package.json:7]: @LukasTy Correct me if I'm wrong, but you mean we will end up with the following internal packages

- `test/unit-date-pickers`
- `test/unit-charts`
- `test/unit-data-grid`
- `test/unit-tree-view`



About community/pro testing, if some pro types are needed to define some test, then you can test both pro and community future in the pro package

[the remark this answers is not in the document]
Comment (LukasTy) [reply test/unit-date-pickers/package.json:7]: Oh, sorry, I forgot to reply to this one. 🙈 
Yes, exactly, I was mentioning the "duplication" or introduction of `test/unit-<package>` packages for each component family. 👍 
I'm not against it, just had a different picture in my head initially and the proposed config is somewhat unconventional. 🤔
